### PR TITLE
[15] changes to build iRODS API example under 4.2.8

### PR DESCRIPTION
**Pull request by d-w-moore**

### CODE REVIEW

**Comment by d-w-moore** [comment]
Not quite ready ...
This is required for the `iapi_example` command to run on Ubuntu16 (not on Ub16 however, so might be a RPATH/RUNPATH thing):
```
export LD_LIBRARY_PATH=/opt/irods-externals/clang-runtime6.0-0/lib
```

**Comment by trel** [comment]
get it down to single commit if it's behaving

**Comment by d-w-moore** [comment]
Down to one commit, leaving un#'d in case anyone has input @SwooshyCueb @alanking 
Summary of changes:

   - exact irods version -> 4.2.8  (currently building & working on 4.2.8 and 4-2-stable)
   - json replacing jansson
   - build path set in CMake, so that iapi_example works from the build directory

File: advanced/irods_api_example/CMakeLists.txt [hunk 1]
```diff
@@ -25,14 +25,20 @@ if (NOT CPACK_PACKAGING_INSTALL_PREFIX)
...
+set(CMAKE_SKIP_RPATH OFF) 
 
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++")
 set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -stdlib=libc++")
 set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -stdlib=libc++")
 add_compile_options(-nostdinc++ -Wall -Wextra -Werror -Wno-unused-parameter)
 link_libraries(c++abi)
-include_directories(${IRODS_EXTERNALS_FULLPATH_CLANG}/include/c++/v1)
+include_directories(${IRODS_EXTERNALS_FULLPATH_CLANG}/include/c++/v1
+${IRODS_EXTERNALS_FULLPATH_JSON}/include )
```
Comment (trel) [inline advanced/irods_api_example/CMakeLists.txt:41]: can you line this up a little better?    not sure our convention on this...

**Comment by trel** [comment]
@d-w-moore can you revisit this in light of 4.2.10?

**Comment by d-w-moore** [comment]
Sure will.

**Comment by trel** [comment]
no longer necessary.

**Pull request closed without merging**